File: MyStagePass.Services/Database/Purchase.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace MyStagePass.Services.Database
{
	public class Purchase
	{
		public int PurchaseID { get; set; }
		public DateTime PurchaseDate { get; set; }
		public virtual Customer Customer { get; set; } = null!;
		public int CustomerID {  get; set; }
		public virtual ICollection<Ticket> Tickets { get; set; } = new List<Ticket>();
		public bool IsDeleted { get; set; } = false;
		public string? PaymentIntentId { get; set; }
		public bool? IsRefunded { get; set; }

		[NotMapped]
		public int Total => Tickets.Sum(i => i.Price);
	}
}

File: UI/mystagepass_mobile/lib/models/Purchase/purchase.dart
import 'package:json_annotation/json_annotation.dart';
import '../Ticket/ticket.dart';

part 'purchase.g.dart';

@JsonSerializable()
class Purchase {
  int? purchaseID;
  DateTime? purchaseDate;
  int? customerID;
  List<Ticket>? tickets;
  bool? isDeleted;
  int? total;
  String? paymentIntentId;
  bool? isRefunded;

  Purchase({
    this.purchaseID,
    this.purchaseDate,
    this.customerID,
    this.tickets,
    this.isDeleted,
    this.total,
    this.isRefunded,
    this.paymentIntentId,
  });

  factory Purchase.fromJson(Map<String, dynamic> json) =>
      _$PurchaseFromJson(json);
  Map<String, dynamic> toJson() => _$PurchaseToJson(this);
}
